aws/resource_aws_sagemaker_notebook_instance_test.go

```diff
@@ -0,0 +1,320 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+const sagemakerTestAccSagemakerNotebookInstanceResourceNamePrefix = "terraform-testacc-"
+
+func init() {
+	resource.AddTestSweepers("aws_sagemaker_notebook_instance", &resource.Sweeper{
+		Name: "aws_sagemaker_notebook_instance",
+	})
+}
+
+func TestAccAWSSagemakerNotebookInstance_basic(t *testing.T) {
+	var notebook sagemaker.DescribeNotebookInstanceOutput
+	notebookName := resource.PrefixedUniqueId(sagemakerTestAccSagemakerNotebookInstanceResourceNamePrefix)
+
+	resource.Test(t, resource.TestCase{
```

Comment:
We can make these parallel by swapping `resource.Test` for `resource.ParallelTest`. This will detect the error I mentioned below where the names aren't randomized